test/applications_integration_test.go

```diff
@@ -785,7 +785,7 @@ func theContentsOfTheBoxWithNameShouldBeIfThereIsAnErrorItIs(fromClient, encoded
 	} else {
 		err = fmt.Errorf("expecting algod or indexer, got " + fromClient)
 	}
-	if err != nil {
+	if err != nil && len(errStr) != 0 {
```

Comment:
I believe you'll also want to error if `len(errStr) != 0` but `err` is nil, since you expected an error that wasn't there

Reply:
good point! Fixed in the latest commits